File: 00. Estructura/01. Lenguajes de proposito general/02. Programación básica/07. Funciones/03. Sin paramatros y con retorno multiple/C++/Sin paramatros y con retorno multiple.cpp

// Estructura de retorno
struct Return {
    tipo return1;
    tipo return2;
    tipo returnN;
};

// Creación
Return funcion() {
    // Contenido
    return (Return) {valor1, valor2, valorN};
}

int main() {
    
    // Llamado
    /*Contexto*/ funcion() /*Contexto*/

    // Llamado en asignación
    Return variables = funcion();
    tipo varible1 = variables.return1;
    tipo varible2 = variables.return2;
    tipo varibleN = variables.returnN;

    // Eliminación
    NO SOPORTADO (SE ELIMINAN AL SALIR DE LOS CORCHETES CONTENEDORES)

}

File: 01. Lenguajes de proposito general/02. Programación básica/07. Funciones/03. Sin paramatros y con retorno multiple/Java/Sin paramatros y con retorno multiple.java

package prueba;

public class Prueba {

    // Creación
    public static Object[] funcion() {
        // Contenido
        return new Object[]{valor1, valor2, valorN};
    }

    public static void main(String[] args) {
        
        // Llamado
        /*Contexto*/ funcion() /*Contexto*/

        // Llamado en asignación
        Object[] variables = funcion();
        tipo varible1 = (tipo) variables[0];
        tipo varible2 = (tipo) variables[1];
        tipo varibleN = (tipo) variables[2];

        // Eliminación
        NO SOPORTADO (EL GARBAGE COLLECTOR LA ELIMINA AUTOMATICAMENTE CUANDO YA ES INACCESIBLE)

    }
    
}